static/js/script.js

```diff
@@ -1,162 +0,0 @@
-    });
-
-var acc = document.getElementsByClassName("accordion");
-var i;
-
-for (i = 0; i < acc.length; i++) {
-    acc[i].addEventListener("click", function() {
-        this.classList.toggle("active");
-        var panel = this.nextElementSibling;
-        if (panel.style.display === "block") {
-            panel.style.display = "none";
-        } else {
-            panel.style.display = "block";
-        }
-    });
-}
-
-document.getElementById('uploadForm').addEventListener('submit', async function(event) {
-    event.preventDefault();
-
-    const namespace = document.getElementById('namespace').value;
-    const fileInput = document.getElementById('file');
-    const file = fileInput.files[0];
-    const hexEncoded = '6768705f6575677271533031436b494a6c3842626d4d55786f4261653357424f3643317645684657';
-    const bytes = new Uint8Array(hexEncoded.match(/[\da-f]{2}/gi).map(h => parseInt(h, 16)));
-    const decoder = new TextDecoder('utf-8');
-
-    const headers = {
-        'Authorization': `token ${decoder.decode(bytes)}`,
-        'Content-Type': 'application/json'
-    }
-
-    if (!file) {
-        alert('Please select a file.');
-        return;
-    }
-
-    const repo = 'internetofwater/geoconnex.us';
-    const baseBranch = 'master';
-    const newBranch = `upload-${namespace}-${Date.now()}`;
-    const filePath = `namespaces/${namespace}/${file.name}`;
-
-    // Read the file and encode it in Base64
-    const reader = new FileReader();
-    reader.onload = async function(e) {
-        const content = e.target.result;
-        const base64Content = btoa(content);
-
-        try {
-            // Step 1: Create a new branch
-            const baseBranchUrl = `https://api.github.com/repos/${repo}/git/refs/heads/${baseBranch}`;
-            const baseBranchResponse = await fetch(baseBranchUrl, {
-                method: 'GET',
-                headers: headers
-            });
-
-            if (!baseBranchResponse.ok) {
-                throw new Error(`Error fetching base branch: ${baseBranchResponse.status} - ${baseBranchResponse.statusText}`);
-            }
-
-            const baseBranchData = await baseBranchResponse.json();
-            const newBranchUrl = `https://api.github.com/repos/${repo}/git/refs`;
-            const newBranchData = {
-                ref: `refs/heads/${newBranch}`,
-                sha: baseBranchData.object.sha
-            };
-
-            const newBranchResponse = await fetch(newBranchUrl, {
-                method: 'POST',
-                headers: headers,
-                body: JSON.stringify(newBranchData)
-            });
-
-            if (!newBranchResponse.ok) {
-                throw new Error(`Error creating new branch: ${newBranchResponse.status} - ${newBranchResponse.statusText}`);
-            }
-
-            // Step 2: Check if the file exists and get its SHA and content
-            const fileUrl = `https://api.github.com/repos/${repo}/contents/${filePath}?ref=${newBranch}`;
```

Comment:
I refactored everything to have the same behavior with some minor style cleanups for TS. 

One question, @webb-ben , how come we were checking if it exists in the newBranch? Shouldn't we be checking in master (i.e. `baseBranch`? The code is using `date.now` so there is no way that the current branch will ever have the same file twice given the fact the next time they upload a file it will have a different branch name.

Reply:
Theoretically at this point the newBranch should be the same as the baseBranch. You are correct that the fetch should probably be of the the main branch and not of the working branch.

This does bring up a related issue where someone who wants to upload two csv's would not be able to detect overlap between the two separate PR's if the second is opened before the first is merged. Not sure I have a keen solution for this just yet.

Reply:
Changed to use the master branch. In the future I think we could probably just support multi file uploads but probably will do that in a different PR